app/src/components/Main/Main.js

```diff
@@ -198,13 +213,18 @@ function Main(props) {
             type="text"
             placeholder="Enter your username"
             name="username"
+            minlength="2"
             value={data.username}
             onChange={validateInput}
           />
           <span className="popup__error">{errors.username}</span>
+          <span className="popup__error popup__error_type_submit">
+            {signUpError}
+          </span>
           <input
             className="popup__submit"
             disabled={signUpState}
+            onClick={props.handleSignUp}
```

Comment:
* Submit handlers are added only to tags `form` with event `submit` rather than to submit buttons with event `click` because it also automatically handles `Enter` presses